Upgrade: use oceanifier@4.0.1 to simplify test/runner.html

cyjake opened the pull request:
I've refactored and enhanced oceanifier a lot. With oceanifier@4.0.1, the `test/runner.html` can be simplified drastically.

cyjake committed:
Upgrade: use oceanifier@4.0.1 to simplify test/runner.html

dotnil commented on test/runner.js:
```diff
@@ -1,4 +1,8 @@
 require('./test.heredoc')
 require('./test.heredoc.strip')
+
+mocha.run()
```
:flushed: didn't test this in Node. I'll see to it asap.

cyjake committed:
Fix: make test/runner.js work in both node and browser

Upgraded oceanifier to 4.0.2 for proper dependencies' cache. Now
oceanifier will compile and concat dependencies into tmp folder as
cache. This reduces requests when test in browser.

dotnil commented on test/runner.html:
```diff
@@ -8,16 +8,6 @@
-    seajs.use('heredoc/test/runner', function() {
-      mocha.run()
-    })
-  </script>
```
This PR is about this change mostly. There's no more strange stuff here. Hope you like it @jden

dotnil commented on test/runner.js:
```diff
@@ -1,4 +1,9 @@
 require('./test.heredoc')
 require('./test.heredoc.strip')
+
+if (typeof mocha !== 'undefined') mocha.run()
```
When running in browser, we need to perform `mocha.setup()` and `mocha.run()`,

dotnil commented:
ping @jden

jden commented:
sorry for the unusually long delay! I've been travelling without regular internet access ^.^

merged. since this is a dev-only change, I haven't republished on npm.
